http/src/main/java/io/crate/http/netty/CrateNettyHttpServerTransport.java

```diff
@@ -76,11 +75,41 @@ public ChannelPipelineFactory configureServerChannelPipelineFactory() {
     }
 
     public void addBefore(ChannelPipelineItem item) {
-        addBeforeList.add(item);
+        synchronized (addBeforeList) {
+            addBeforeSorted(addBeforeList, item);
+        }
+    }
+
+    List<ChannelPipelineItem> addBefore() {
+        return addBeforeList;
     }
 
-    public void addAfter(ChannelPipelineItem item) {
-        addAfterList.add(item);
+    private void addBeforeSorted(List<ChannelPipelineItem> addBeforeList, ChannelPipelineItem newItem) {
+        addBeforeList.add(newItem);
+
+        ArrayList<ChannelPipelineItem> copy = new ArrayList<>(addBeforeList.size());
+        copy.addAll(addBeforeList);
+
+        for (ChannelPipelineItem item : copy) {
+            addBeforeList.remove(item);
+
+            Optional<ChannelPipelineItem> prev = addBeforeList.stream()
+                .filter(o -> o.name.equals(item.base))
+                .findFirst();
```

Comment:
using `stream().filter().findFirst()` and then again `indexOf() ` always has to loop twice over the collection to find the entry. I think this could be done more efficient and as easy without using the stream API. But since this is only executed on startup I guess it's okay this way too.

On a different note - couldn't this be done with `List.sort()` and a custom comparator?